pageserver/src/http/openapi_spec.yml

```diff
@@ -429,7 +429,11 @@ paths:
               schema:
                 $ref: "#/components/schemas/SyntheticSizeResponse"
             text/html:
-              description: SVG representation of the tenant and it's timelines.
+              schema:
+                type: string
+              examples:
+                html:
+                  summary: SVG representation of the tenant and it's timelines.
```

Comment:
Hm, after running another linter I got this output
```
 validating /home/runner/work/cloud/cloud/swagger/pageserver.yml...
[1] ../swagger/pageserver.yml:436:19 at #/paths/~1v1~1tenant~1{tenant_id}~1synthetic_size/get/responses/200/content/text~1html/examples/html/value

Example value must conform to the schema: type must be string.

434 |       examples:
435 |         html:
436 |           summary: SVG representation of the tenant and it's timelines.
437 | "401":
438 |   description: Unauthorized Error

referenced from ../swagger/pageserver.yml:432:15 at #/paths/~1v1~1tenant~1{tenant_id}~1synthetic_size/get/responses/200/content/text~1html 

Error was generated by the no-invalid-media-type-examples rule.


[2] ../swagger/pageserver.yml:790:5 at #/components/schemas/TenantLoadRequest

Component: "TenantLoadRequest" is never used.

788 |           type: integer
789 |           description: Attachment generation number.
790 | TenantLoadRequest:
791 |   type: object
792 |   properties:

Error was generated by the no-unused-components rule.


[3] ../swagger/pageserver.yml:380:3 at #/paths/~1v1~1tenant~1{tenant_id}~1{timeline_id}~1preserve_initdb_archive

Paths should resolve unambiguously. Found two ambiguous paths: `/v1/tenant/{tenant_id}/timeline/{timeline_id}` and `/v1/tenant/{tenant_id}/{timeline_id}/preserve_initdb_archive`.

378 |               $ref: "#/components/schemas/ConflictError"
379 |
380 | /v1/tenant/{tenant_id}/{timeline_id}/preserve_initdb_archive:
381 |   parameters:
382 |     - name: tenant_id

Warning was generated by the no-ambiguous-paths rule.
```

Can you please look into them too?

Reply:
Just in case, linter is `redocly lint --config=redocly.yaml` with this config
```yaml
organization: Neon
apis:
  pageserver@v1:
    root: swagger/pageserver.yml (this should be just your path, I think)
    rules:
      operation-2xx-response: off
      operation-operationId: off
      no-path-trailing-slash: off

rules:
  no-unused-components: error
  no-invalid-media-type-examples: error
  operation-4xx-response: off
  operation-summary: off
  info-license-url: off
  operation-2xx-response: error
extends:
  - recommended

```

Reply:
Thanks for the review.

I tried `redocly lint`, but I did not encounter `no-invalid-media-type-example` error.
```
$ redocly --version
1.17.1
$ redocly lint --config=.vscode/redocly.yaml
validating /pageserver/src/http/openapi_spec.yml...
[1] pageserver/src/http/openapi_spec.yml:790:5 at #/components/schemas/TenantLoadRequest

Component: "TenantLoadRequest" is never used.

788 |           type: integer
789 |           description: Attachment generation number.
790 | TenantLoadRequest:
791 |   type: object
792 |   properties:

Error was generated by the no-unused-components rule.


[2] pageserver/src/http/openapi_spec.yml:380:3 at #/paths/~1v1~1tenant~1{tenant_id}~1{timeline_id}~1preserve_initdb_archive

Paths should resolve unambiguously. Found two ambiguous paths: `/v1/tenant/{tenant_id}/timeline/{timeline_id}` and `/v1/tenant/{tenant_id}/{timeline_id}/preserve_initdb_archive`.

378 |               $ref: "#/components/schemas/ConflictError"
379 |
380 | /v1/tenant/{tenant_id}/{timeline_id}/preserve_initdb_archive:
381 |   parameters:
382 |     - name: tenant_id

Warning was generated by the no-ambiguous-paths rule.


/pageserver/src/http/openapi_spec.yml: validated in 78ms

❌ Validation failed with 1 error and 1 warning.
run `redocly lint --generate-ignore-file` to add all problems to the ignore file.
```

> [1] ../swagger/pageserver.yml:436:19 at #/paths/~1v1~1tenant~1{tenant_id}~1synthetic_size/get/responses/200/content/text~1html/examples/html/value

However, I found a mistake and have since corrected it. Could you please verify it?
Fixed in commit https://github.com/neondatabase/neon/pull/8273/commits/ffee8e6b297e375152e1e040d6c492050db24ac1 .


> [2] ../swagger/pageserver.yml:790:5 at #/components/schemas/TenantLoadRequest

The schema `TenantLoadRequest` was not utilized anywhere, so it has been fixed in commit https://github.com/neondatabase/neon/pull/8273/commits/7b05668cb8b6b5ef0e82a16c7e0abe718e0d93e5


> [3] ../swagger/pageserver.yml:380:3 at #/paths/~1v1~1tenant~1{tenant_id}~1{timeline_id}~1preserve_initdb_archive

The API path is ambiguous, so I have made corrections in commit https://github.com/neondatabase/neon/pull/8273/commits/3dbb970fa834d7bf9e48508c5d35c1ef5c5d24c7 .
However, this change is incompatible because API path changed, it's up to you to push it or not.